File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/producers/SingleBookProducer.java
package ru.kmorozov.gbd.core.producers;

import ru.kmorozov.gbd.core.logic.context.IBookListProducer;

import java.util.Collections;
import java.util.HashSet;
import java.util.Set;

/**
 * Created by km on 12.11.2016.
 */
public class SingleBookProducer implements IBookListProducer {

    private final Set<String> singletonSet;

    public SingleBookProducer(String bookId) {
        this.singletonSet = new HashSet<>(Collections.singletonList(bookId));
    }

    @Override
    public Set<String> getBookIds() {
        return this.singletonSet;
    }
}

File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/producers/SingleBookProducer.kt
package ru.kmorozov.gbd.core.producers

import ru.kmorozov.gbd.core.logic.context.IBookListProducer
import java.util.*

/**
 * Created by km on 12.11.2016.
 */
class SingleBookProducer(bookId: String) : IBookListProducer {

    override val bookIds: Set<String>

    init {
        bookIds = HashSet(listOf(bookId))
    }
}
